app/models/address.rb

```diff
@@ -145,6 +147,20 @@ def geocode_best_possible
 
   end
 
+  # TODO: Make this a private method after a one-time rake task - which updates
+  #       non-conforming city names in addresses - has been run.
+  def format_city_name
```

Comment:
This method is great.  But it doesn't belong in the Address model.  this should happen before it gets to the Address model.
This should be in a helper: this will fix-up validate the address data.
This will also make the method available to any views that need to validate/fix the city name

Reply:
@weedySeaDragon I don't agree.  The intent is to make sure that any city name conforms to our standard format.  It's not intended to be used in views because we should _never_ need it, since the city will be correctly formatted in the DB.  Also, we're not _validating_ the city and then showing the corrected version to the user (and asking them to mimic that corrected form).  We'll just fix it for them and be done with it.

The only intended use outside of that is a one-time conversion of `city` in the DB.

Reply:
This increases the _responsibiltiy_ for the Address class (which already is waaaay too big with multiple responsibilities).   This violates the Single Responsibility Principle by adding yet another responsibility for the Address class.

I agree that it doesn't need to be in a helper class.  So it should be a simple `Service` class (or some other name that communicates that it is not part of the ActiveRecord hierarchy and just provides a service or behavior).

This class has 1 simple responsibility:  Ensure that a city name is formatted correctly and is a valid city name.

Reply:
@weedySeaDragon I've taken another look and thought about this more and decided that I want to retain that method within the Address class.  It has no other purpose than checking format - and refomatting if required - a specific attribute of instances of the class.  To break it out in its own class would be an exercise in defining classes on a too-granular basis, IMO.  And, the method is entirely consistent with the overall responsibility of the containing class.

We'll just have to disagree on this one.